### DISCUSSION TITLE
Bug: eslint v9.0 peer dependency issues

### DISCUSSION THREAD

**Discussion by marijaduvnjak** [eslint/eslint#18343] [Q&A] [answered]
### Environment

Node version: v20.12.2
npm version: v10.5.0
Local ESLint version: v8.57.0 (Currently used)
Global ESLint version: Not found
Operating System: linux 6.8.0-76060800daily20240311-generic

### What parser are you using?

@typescript-eslint/parser

### What did you do?

I tried to install the latest eslint `v9.0.0` version in my nextjs project.

### What did you expect to happen?

I expected that I could install the latest eslint without any issues.

### What actually happened?

Instead, I ran into peer dependency issues during the installation:
```bash
Already up to date
Progress: resolved 424, reused 416, downloaded 0, added 0, done
 WARN  Issues with peer dependencies found
.
├─┬ @typescript-eslint/eslint-plugin 7.6.0
│ ├── ✕ unmet peer eslint@^8.56.0: found 9.0.0
│ └─┬ @typescript-eslint/type-utils 7.6.0
│   ├── ✕ unmet peer eslint@^8.56.0: found 9.0.0
│   └─┬ @typescript-eslint/utils 7.6.0
│     └── ✕ unmet peer eslint@^8.56.0: found 9.0.0
├─┬ @typescript-eslint/parser 7.6.0
│ └── ✕ unmet peer eslint@^8.56.0: found 9.0.0
├─┬ eslint-config-airbnb-typescript 18.0.0
│ ├── ✕ unmet peer eslint@^8.56.0: found 9.0.0
│ └─┬ eslint-config-airbnb-base 15.0.0
│   └── ✕ unmet peer eslint@"^7.32.0 || ^8.2.0": found 9.0.0
├─┬ eslint-config-next 14.2.1
│ ├── ✕ unmet peer eslint@"^7.23.0 || ^8.0.0": found 9.0.0
│ ├─┬ @typescript-eslint/parser 6.21.0
│ │ └── ✕ unmet peer eslint@"^7.0.0 || ^8.0.0": found 9.0.0
│ ├─┬ eslint-plugin-import 2.29.1
│ │ └── ✕ unmet peer eslint@"^2 || ^3 || ^4 || ^5 || ^6 || ^7.2.0 || ^8": found 9.0.0
│ ├─┬ eslint-plugin-jsx-a11y 6.8.0
│ │ └── ✕ unmet peer eslint@"^3 || ^4 || ^5 || ^6 || ^7 || ^8": found 9.0.0
│ ├─┬ eslint-plugin-react 7.34.1
│ │ └── ✕ unmet peer eslint@"^3 || ^4 || ^5 || ^6 || ^7 || ^8": found 9.0.0
│ └─┬ eslint-plugin-react-hooks 4.6.0
│   └── ✕ unmet peer eslint@"^3.0.0 || ^4.0.0 || ^5.0.0 || ^6.0.0 || ^7.0.0 || ^8.0.0-0": found 9.0.0
├─┬ eslint-plugin-import 2.29.1
│ └── ✕ unmet peer eslint@"^2 || ^3 || ^4 || ^5 || ^6 || ^7.2.0 || ^8": found 9.0.0
└─┬ eslint-watch 8.0.0
  └── ✕ unmet peer eslint@">=8 <9.0.0": found 9.0.0

Done in 3.6s
```

### Link to Minimal Reproducible Example

https://codesandbox.io/p/devbox/eslint-v9-0-peer-dependency-issues-vmyx88?file=%2Fpackage.json

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by kecrily** [comment, marked as answer]
This is because the version of the shareable configuration or plugin you are using does not currently support v9. You can try updating them to the latest version and keep an eye on v9 support for related dependencies

https://github.com/typescript-eslint/typescript-eslint/issues/8211
